hazelcast/src/main/java/com/hazelcast/sql/SqlResult.java

```diff
@@ -41,33 +41,24 @@
  * }
  * </pre>
  *
- * <p>
  * <h4>Usage for update count</h4>
  *
  * <pre>
  *     long updated = hazelcastInstance.getSql().execute("UPDATE ...").updateCount();
  * </pre>
  *
  * You don't need to call {@link #close()} in this case.
+ *
+ * <h4>Determining the type of the result</h4>
+ *
+ * To determine whether the result contains rows or an update count, call
+ * {@link #updateCount()}. If it returns -1, the result is rows, otherwise
+ * it's update count.
  */
 public interface SqlResult extends Iterable<SqlRow>, AutoCloseable {
 
     /**
-     * If this result represents a row set, this method returns {@code false}.
-     * If this result represents an update count (such as for a DML query), it
-     * returns {@code true}.
-     *
-     * @return {@code false} for a rows result and {@code true} for an update
-     *     count result
-     */
-    boolean isUpdateCount();
-
-    /**
-     * Gets row metadata.
-     *
-     * @throws IllegalStateException if this result doesn't have rows (i.e.
-     *     when {@link #isUpdateCount()} returns {@code true})
-     * @return row metadata
+     * Gets the row metadata or {@code null} if this result is an update count.
```

Comment:
What is the motivation for returning `null`? It is inconsistent with the behavior of the `iterator()` method. and makes the code prone to NPE. The previous behavior does not have this problem.

Reply:
It's stale Javadoc, I first returned `null` but changed it for the same reason. Fixed.